internal/controller/scaledobject_controller.go

```diff
@@ -0,0 +1,1110 @@
+	name := metadata["name"]
+	value := metadata["value"]
+
+	if name == "" || value == "" {
+		return nil, fmt.Errorf("kubernetes-workload trigger missing required fields")
+	}
+
+	targetValue, err := resource.ParseQuantity(value)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kubernetes-workload value: %w", err)
+	}
+
+	metric := &kpav1alpha1.MetricSpec{
+		Type: kpav1alpha1.ObjectMetricType,
+		Object: &kpav1alpha1.ObjectMetricSource{
+			DescribedObject: kpav1alpha1.CrossVersionObjectReference{
+				Kind:       "Pod",
+				Name:       name,
+				APIVersion: "v1",
+			},
+			Metric: kpav1alpha1.MetricIdentifier{
+				Name: "pending_pods",
+			},
+			Target: kpav1alpha1.MetricTarget{
+				Type:  kpav1alpha1.ValueMetricType,
+				Value: &targetValue,
+			},
+		},
+	}
+
+	// Use simple selector referencing the ScaledObject
+	metric.Object.Metric.Selector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"scaledobject.keda.sh/name": scaledObjectName,
+		},
+	}
+
+	return metric, nil
+}
+
+// translateExternalTrigger handles generic external trigger translation
+func (r *ScaledObjectReconciler) translateExternalTrigger(metadata map[string]string, triggerName string, scaledObjectName string) (*kpav1alpha1.MetricSpec, error) {
+	metricName := metadata["metricName"]
+	if metricName == "" {
+		metricName = triggerName
+	}
+
+	targetValue := metadata["targetValue"]
+	if targetValue == "" {
+		return nil, fmt.Errorf("external trigger missing targetValue")
+	}
+
+	targetQuantity, err := resource.ParseQuantity(targetValue)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse targetValue: %w", err)
+	}
+
+	metric := &kpav1alpha1.MetricSpec{
+		Type: kpav1alpha1.ExternalMetricType,
+		External: &kpav1alpha1.ExternalMetricSource{
+			Metric: kpav1alpha1.MetricIdentifier{
+				Name: metricName,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"scaledobject.keda.sh/name": scaledObjectName,
+					},
+				},
+			},
+			Target: kpav1alpha1.MetricTarget{
+				Type:  kpav1alpha1.ValueMetricType,
+				Value: &targetQuantity,
+			},
+		},
+	}
+
+	return metric, nil
+}
+
+// translateExternalScalerTrigger handles specific external scaler triggers (redis, kafka, etc.)
+func (r *ScaledObjectReconciler) translateExternalScalerTrigger(scalerType string, metadata map[string]string, scaledObjectName string) (*kpav1alpha1.MetricSpec, error) {
+	// Generic handling for external scalers
+	// Each scaler type would have its specific metadata requirements
+
+	// Generate metric name similar to KEDA pattern
+	var metricName string
+	var targetValue string
+
+	switch scalerType {
+	case "redis":
+		targetValue = metadata["listLength"]
+		listName := metadata["listName"]
+		if listName != "" {
+			metricName = fmt.Sprintf("s0-%s-%s", scalerType, listName)
+		} else {
+			metricName = fmt.Sprintf("s0-%s", scalerType)
+		}
```

Comment:
Does it make sense to extract the logic to smaller function to handle metricName generation, for readability and clarity purposes.